pkgs/checks/test/extensions/collection_equality_test.dart

```diff
@@ -129,9 +129,9 @@ void main() {
       }, {
         'a': (Subject<dynamic> it) => it.isA<String>().startsWith('a')
       })).isNotNull().deepEquals([
-        "has no entry to match 'a': <A value that:",
-        '  is a String',
-        "  starts with 'a'>",
+        "has an element at ['a'] that:",
```

Comment:
It seems a bit unfortunate that we lose the expectation text here, is it just too noisy with that as well as the actual? Or a technical limitation?

Reply:
The expectation is the "which does not start with 'a'" at the end, right? Or is there a different thing you have in mind as the expectation text?

Reply:
The expectation text was `is a String starts with 'a'`. In this version we only get the part describing why it didn't match, but not what was expected. In this case, you can still reason about it based on only the mismatch text but I don't think that would always be the case.

Reply:
This test covers the "which" part of the failure. The `is a String starts with 'a'` content is still present in the "expected" part of the full failure, it's jut not repeated in the "which" like it used to be.

The full old failure was:

```
  Expected: a Map<String, String> that:
    is deeply equal to {'a': A value that:
      is a String
      starts with 'a'}
  Actual: {'a': 'b'}
  Which: has no entry to match 'a': <A value that:
    is a String
    starts with 'a'>
```

Here is the new full failure:

```
  Expected: a Map<String, String> that:
    is deeply equal to {'a': A value that:
      is a String
      starts with 'a'}
  Actual: {'a': 'b'}
  Which: has an element at ['a'] that:
    Actual: 'b'
    which does not start with 'a'
```

Does that seem sufficient?

Reply:
In this case the `matcher` output is better than we can get with `checks`. The difference largely comes from the fact that with `matcher` composing multiple expectations uses utilities like `allOf` instead of chaining expectations with cascades. I spent some time trying to figure out whether we could output more compact messages from `checks` in cases with a single expectation but I didn't get anywhere with it.

```
  Expected: {'a': <a string starting with 'a'>}
    Actual: {'a': 'b'}
     Which: at location ['a'] is 'b' which does not match a string starting with 'a'
```

Reply:
LGTM